File: Backend/Src/Models/Report.cs
using System.ComponentModel.DataAnnotations;

namespace Intercab.Models;

public partial class Report
{
	public int Id { get; set; }

	[MinLength(10)]
	public required string Name { get; set; }

	public required string Path { get; set; }

	public sbyte IsGenerated { get; set; }

	public DateTime CreatedDate { get; set; }

	public int CompanyId { get; set; }

	public virtual Company? Company { get; set; }

	public virtual ICollection<Trip> Trips { get; set; } = [];
}

File: Frontend/src/models/Report.ts
import { Company } from "./Company";
import { Trip } from "./Trip";

export interface Report {
	id: number;
	name: string;
	path: string;
	isGenerated: number;
	createdDate?: string;
	companyId: number;
	company?: Company;
	trips?: Trip[];
}
